src/main/java/appeng/recipes/handlers/Inscribe.java

```diff
@@ -63,7 +64,12 @@ public void register() throws RegistrationError, MissingIngredientError
 		final ItemStack output = this.getOutput().getItemStack();
 		final InscriberProcessType type = InscriberProcessType.INSCRIBE;
 
-		final IInscriberRecipe recipe = new InscriberRecipe( inputs, output, top, bot, type );
+		final IInscriberRecipe recipe = builder.withInputs( inputs )
+				.withOutput( output )
+				.withTopOptional( top )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Null passed for non-null parameter of appeng.api.features.IInscriberRecipeBuilder.withTopOptional(ItemStack) in appeng.recipes.handlers.Inscribe.register() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ANP_NULL_PARAM_DEREF)